File: src/RealWorld.Application/DTOs/UserWithToken.cs
namespace RealWorld.Application.DTOs;

public class UserWithToken
{
    public string Email { get; set; }
    public string Username { get; set; }
    public string Bio { get; set; }
    public string Image { get; set; }
    public string Token { get; set; }

    public UserWithToken(UserData userData, string token)
    {
        Email = userData.Email;
        Username = userData.Username;
        Bio = userData.Bio;
        Image = userData.Image;
        Token = token;
    }
}

File: src/main/java/io/spring/application/data/UserWithToken.java
package io.spring.application.data;

import lombok.Getter;

/** Composite DTO combining user profile data with the current JWT for API responses. */
@Getter
public class UserWithToken {
  private String email;
  private String username;
  private String bio;
  private String image;
  private String token;

  public UserWithToken(UserData userData, String token) {
    this.email = userData.getEmail();
    this.username = userData.getUsername();
    this.bio = userData.getBio();
    this.image = userData.getImage();
    this.token = token;
  }
}
